src/server/editorServices.ts

```diff
@@ -1105,187 +1312,159 @@ namespace ts.server {
+            project.setCompilerOptions(newOptions);
             // VS only set the CompileOnSaveEnabled option in the request if the option was changed recently
             // therefore if it is undefined, it should not be updated.
             if (compileOnSave !== undefined) {
                 project.compileOnSaveEnabled = compileOnSave;
             }
-            project.setProjectErrors(concatenate(configFileErrors, projectErrors));
-
-            project.updateGraph();
+            this.addFilesToNonInferredProjectAndUpdateGraph(project, newUncheckedFiles, propertyReader, /*clientFileName*/ undefined, newTypeAcquisition, configFileErrors);
         }
 
-        private updateConfiguredProject(project: ConfiguredProject) {
-            if (!this.host.fileExists(project.getConfigFilePath())) {
-                this.logger.info("Config file deleted");
-                this.removeProject(project);
-                return;
-            }
+        /**
+         * Read the config file of the project again and update the project
+         * @param project
+         */
+        /* @internal */
+        reloadConfiguredProject(project: ConfiguredProject) {
+            // At this point, there is no reason to not have configFile in the host
 
             // note: the returned "success" is true does not mean the "configFileErrors" is empty.
             // because we might have tolerated the errors and kept going. So always return the configFileErrors
             // regardless the "success" here is true or not.
-            const { success, projectOptions, configFileErrors } = this.convertConfigFileContentToProjectOptions(project.getConfigFilePath());
-            if (!success) {
-                // reset project settings to default
-                this.updateNonInferredProject(project, [], fileNamePropertyReader, {}, {}, /*compileOnSave*/ false, configFileErrors);
-                return configFileErrors;
+            const host = project.getCachedServerHost();
+            host.clearCache();
+            const configFileName = project.getConfigFilePath();
+            this.logger.info(`Reloading configured project ${configFileName}`);
+            const { projectOptions, configFileErrors, configFileSpecs } = this.convertConfigFileContentToProjectOptions(configFileName, host);
+            project.configFileSpecs = configFileSpecs;
+            this.updateConfiguredProject(project, projectOptions, configFileErrors);
+
+            if (!this.eventHandler) {
+                return;
             }
 
+            this.eventHandler(<ConfigFileDiagEvent>{
+                eventName: ConfigFileDiagEvent,
+                data: { configFileName, diagnostics: project.getGlobalProjectErrors() || [], triggerFile: configFileName }
+            });
+        }
+
+        /**
+         * Updates the configured project with updated config file contents
+         * @param project

```

Comment:
An `@param` annotation is not useful if there are no param-specific comments to add.